src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -32,7 +32,7 @@ private void InitSort(CultureInfo culture)
             }
             else
             {
-                _isAsciiEqualityOrdinal = (_sortName == "en-US" || _sortName == "");
+                _isAsciiEqualityOrdinal = _sortName == "" || _sortName == "en" || (_sortName.Length > 3 && _sortName[0] == 'e' && _sortName[1] == 'n' && _sortName[2] == '-');
```

Comment:
Nit: Comment would be useful on why we are doing this instead of just `StartsWith`. And it would be a bit smaller code to share more between the two cases.

```suggestion
	                // Inline the following condition to avoid potential implementation cycles within globalization
	                //
	                // _isAsciiEqualityOrdinal = _sortName == "" || _sortName == "en" || _sortName.StartsWith("en-", StringComparison.Ordinal);
	                //
	                _isAsciiEqualityOrdinal = _sortName.Length == 0 ||
	                    (_sortName.Length >= 2 && _sortName[0] == 'e' && _sortName[1] == 'n' && (_sortName.Length == 2 || _sortName[2] == '-'));
```